src/users-list.tsx

```diff
@@ -1,95 +1,42 @@
-import React, { useEffect, useState } from 'react';
-import { ApolloClient, gql, InMemoryCache, createHttpLink } from '@apollo/client';
-import { setContext } from '@apollo/client/link/context';
+import React, { useState, useEffect } from 'react';
+import { getUsers, UsersList } from './components/users';
+import { buttonStyles } from './components/login';
 
-interface UserProps {
-  name: string;
-  email: string;
-}
-
-interface UsersListProps {
-  list: Array<UserProps>;
-}
-
-const User: React.FC<UserProps> = (props) => {
-  return (
-    <li>
-      <p>{props.name}</p>
-      <p>{props.email}</p>
-    </li>
-  );
-};
-
-const UsersList: React.FC<UsersListProps> = (props) => {
-  return (
-    <ul>
-      {props.list.map((p) => (
-        <User email={p.email} name={p.name} key={p.name} />
-      ))}
-    </ul>
-  );
-};
-
-const httpLink = createHttpLink({
-  uri: 'https://tq-template-server-sample.herokuapp.com/graphql',
-});
-
-const authLink = setContext((_, { headers }) => {
-  // get the authentication token from local storage if it exists
-  const token = localStorage.getItem('@token');
-  // return the headers to the context so httpLink can read them
-  return {
-    headers: {
-      ...headers,
-      authorization: token ?  `${token}` : ' ',
-    }
-  }
-});
-
-const client = new ApolloClient({
-  cache: new InMemoryCache(),
-  link : authLink.concat(httpLink),
-});
-
-function getUsers() :Promise<string>  {
-  return client.query({
-    query: gql`
-      query Users {
-        users {
-          nodes {
-            id
-            name
-            email
-          }
-        }
-      }
-    `,
-  })
-    .then(result => {
-      return JSON.stringify(result);
-    })
-    .catch(error => {
-      return JSON.stringify(error);
-    });
-}
+const usersPerPage = 10 ;
 
 export const UsersListPage: React.FC = () => {
   const [users, setUsers] = useState([]);
-
+  const [page, setPage] = useState(0);
+  const [hasNextPage, setHasNextPage] = useState(true);
+  
   async function setUsersList () {
-    const json = JSON.parse(await getUsers())
-    setUsers(json.data.users.nodes)
-    console.warn(json)
+    try{
+      const json = JSON.parse(await getUsers(page, usersPerPage));
+      setUsers(users.concat(json.data.users.nodes));
+      !(json.data.users.pageInfo.hasNextPage) ? setHasNextPage(false) : null;
+      console.warn(json.data.users.pageInfo.hasNextPage);
+      setPage(page + usersPerPage);
```

Comment:
aqui não seria assim? 
```suggestion
      setPage(page + 1);
```